app/jira/JIRA.scala

```diff
@@ -1,43 +1,65 @@
 package jira
 
+import java.time.OffsetDateTime
+import java.time.format.DateTimeFormatter
+
 import cats.data.Kleisli
-import cats.syntax.option._
 import cats.instances.future._
-
+import cats.syntax.option._
 import models.Deployment
 import models.DeploymentResult.{Cancelled, Failed, Succeeded}
-import play.api.libs.json.{JsValue, Json}
+import play.Logger
 import play.api.libs.json.Json.obj
-import play.api.libs.ws.{WSAuthScheme, WSClient, WSResponse}
+import play.api.libs.json._
+import play.api.libs.ws.{WSAuthScheme, WSClient}
 
-import scala.concurrent.Future
 import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 
 object JIRA {
 
   case class Context(wsClient: WSClient,
                      browseTicketsUrl: String,
-                     createIssueApiUrl: String,
+                     issueApiUrl: String,
                      username: String,
                      password: String)
 
-  def createIssueIfPossible(deployment: Deployment): Kleisli[Future, Context, Option[WSResponse]] = {
+  case class CreateIssueKey(key: String)
+
+  implicit val issueReads = Json.reads[CreateIssueKey]
+
+  def createIssueIfPossible(deployment: Deployment): Kleisli[Future, Context, Option[CreateIssueKey]] = {
     deployment.jiraComponent match {
-      case Some(jiraComponent) => createIssue(deployment, jiraComponent).map(_.some)
-      case None                => Kleisli.pure[Future, Context, Option[WSResponse]](None)
+      case Some(jiraComponent) => {
+
+        for {
+          issueKey <- createIssue(deployment, jiraComponent)
+          _        <- JiraTransitions.transition(issueKey, "Standard Change Approved")
+          _        <- JiraTransitions.transition(issueKey, "Implemented")
+        } yield issueKey
+
+      }
+      case None => Kleisli.pure[Future, Context, Option[CreateIssueKey]](None)
     }
   }
 
-  def createIssue(deployment: Deployment, jiraComponent: String) = Kleisli[Future, Context, WSResponse] { ctx =>
-    val json = buildPayload(deployment, jiraComponent)
-    ctx.wsClient
-      .url(ctx.createIssueApiUrl)
-      .withAuth(ctx.username, ctx.password, WSAuthScheme.BASIC)
-      .withHeaders("Content-Type" -> "application/json")
-      .post(json)
+  def createIssue(deployment: Deployment, jiraComponent: String) = Kleisli[Future, Context, Option[CreateIssueKey]] {
```

Comment:
Very minor point, but the naming of these methods is now potentially confusing. `createIssue` only does the issue creation, while `createIssueIfPossible` does both issue creation and transition.

Maybe have 
* `def createIssue(deployment, jiraComponent)`
* `def createAndTransitionIssue(deployment, jiraComponent` and a 
* `def createAndTransitionIssueIfPossible(deployment)`

? Less verbose naming suggestions welcome!